packages/bpk-component-page-indicator/src/BpkPageIndicator.js

```diff
@@ -0,0 +1,190 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withButtonAlignment, withRtlSupport } from 'bpk-component-icon';
+import { useSwipeable } from 'react-swipeable';
+import BpkButton from 'bpk-component-button';
+import LeftArrowIcon from 'bpk-component-icon/lg/chevron-left';
+import RightArrowIcon from 'bpk-component-icon/lg/chevron-right';
+import { isRTL, cssModules } from 'bpk-react-utils';
+
+import STYLES from './BpkPageIndicator.module.scss';
+
+const getClassName = cssModules(STYLES);
+const AlignedLeftArrowIcon = withButtonAlignment(withRtlSupport(LeftArrowIcon));
+const AlignedRightArrowIcon = withButtonAlignment(
+  withRtlSupport(RightArrowIcon),
+);
+const DISPLAYED_TOTAL = 5;
+const START_SCROLL_INDEX = Math.floor(DISPLAYED_TOTAL / 2);
+
+export type Props = {
+  className: ?string,
+  showNav?: boolean,
+  currentItem: number,
+  totalItems: number,
+  updateItemCallback?: Function,
+  itemClassName?: string,
+  activeItemClassName?: string,
+  ariaLabel?: string,
+};
+
+const KEYS = {
+  ENTER: 'Enter',
+  SPACE: 'Spacebar',
+};
+
+const BpkPageIndicator = (props: Props) => {
+  const {
+    activeItemClassName,
+    ariaLabel,
+    className,
+    currentItem,
+    itemClassName,
+    showNav,
+    totalItems,
+    updateItemCallback,
+    ...rest
+  } = props;
+  const classNames = getClassName('bpk-page-indicator', className);
+
+  const updateItem = (index) => {
+    let newItem = index;
+    if (index < 0) {
+      newItem = totalItems - 1;
+    }
+    if (index >= totalItems) {
+      newItem = 0;
+    }
+    updateItemCallback(newItem);
+  };
+
+  const handleKeyboardEvent = (e, index) => {
+    if (e.key === KEYS.ENTER || e.key === KEYS.SPACE || e.key === ' ') {
```

Comment:
What does `e.key === '  '` represent here?

Reply:
This is for the onclick function can also be triggered when the user press on a space.
I don't know why just adding key.spacebar didn't seem to work, so I took a page from [here](https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-split-input/src/BpkSplitInput.js#L165) and added `e.key === ' '` to determine